### PR TITLE
improve arxiv references

### LINKED ISSUE #9: arXiv.org citations missing info
__Expected output:__
The [ACL formatting requirements](https://acl-org.github.io/ACLPUB/formatting.html) give the following example for how to cite an eprint on arXiv:

> Mohammad Sadegh Rasooli and Joel R. Tetreault. 2015. [Yara parser: A fast and accurate dependency parser](http://arxiv.org/abs/1503.06733). Computing Research Repository, arXiv:1503.06733. Version 2.

__Actual output:__

The latex template provided in this repo (using the bibtex citation that arXiv provides) gives the following:

> Mohammad Sadegh Rasooli and Joel R. Tetreault. 2015. [Yara parser: A fast and accurate dependency parser](http://arxiv.org/abs/1503.06733).

It omits the eprint prefix and eprint code.

This seems not good, since the resulting reference has no mention of where the citation is from (without following the hyperlink).

----

__Details:__

If one copies the autogenerated bibtex citation from the "Export Bibtex Citation" link [on arXiv](http://arxiv.org/abs/1503.06733), one gets the following: 

```
@misc{arxiv.1503.06733,
  doi = {10.48550/ARXIV.1503.06733},
  url = {https://arxiv.org/abs/1503.06733},
  author = {Rasooli, Mohammad Sadegh and Tetreault, Joel},
  title = {Yara Parser: A Fast and Accurate Dependency Parser},
  publisher = {arXiv},
  year = {2015}
}
```

I notice the bibliography style file `acl_natbib.bst` has specific code to deal with arXiv references:

https://github.com/acl-org/acl-style-files/blob/2445635ea39619b1fcf6eade762d518bee0d3af6/latex/acl_natbib.bst#L259-L260

but, for some reason (I can't grok .bst files well enough to understand why) the output doesn't comply with the recommended format from the formatting requirements.  Instead, using the provided latex template and bst file, the resulting PDF output just has the name, date, and hyperlinked title.  

Adding a note field to the bibtex entry, like
```
note = {Computing Research Repository, {arXiv}:1503.06733. Version 2},
```
would make the output like the recommended, but it seems like the style sheet should do something like this (at least it should include the `{arXiv}:1503.06733` part in the output, I think).  Is there a reason why the bst style sheet doesn't do something like this by default?

### PR DESCRIPTION
**Pull request by LSinev**
Improves work with bib files got from arXiv directly.

Works with `@misc` type only.
Supports `archivePrefix` or `eprinttype`.
Sets `arXiv preprint` if `publisher = {arXiv}` and no eprint number is provided.
So it works with arXiv-produced bib (more info at https://tex.stackexchange.com/a/679743/79756), and some bibs from doi2bib.org created from arxiv doi.

No support for `primaryClass` (no example from ACL), no detection of versions from url (seems too hard for me), and no support for `@article` with `eprint` from DBLP (too much to test for now — may become worse).

Files to check and view improvements
result

[bib_experiment_after.pdf](https://github.com/acl-org/acl-style-files/files/13814079/bib_experiment_after.pdf)
[bib_experiment_before.pdf](https://github.com/acl-org/acl-style-files/files/13814080/bib_experiment_before.pdf)
[acl_natbib_new202401_test.zip](https://github.com/acl-org/acl-style-files/files/13814105/acl_natbib_new202401_test.zip)

May fix #9 

Invite to check/review/approve: @nschneid @davidweichiang @danielgildea @postylem 

P.S. At the start, this addition was inspired by `output.eprint` of `ACM-Reference-Format.bst` from [acmart](https://www.ctan.org/tex-archive/macros/latex/contrib/acmart) but diverged later.

### CODE REVIEW

**Comment by davidweichiang** [comment]
Thanks! Is there any hope of seeing a diff of the changes that isn't the entire bst file?

**Comment by LSinev** [comment]
> entire bst file

wow! github... i'll check, may be some LF <-> CRLF line endings conversion happened

**Comment by LSinev** [comment]
For ease of review, rebased (and force pushed) branch with changes.

**Comment by davidweichiang** [comment]
Thanks! Is there a consensus that "arXiv preprint, arXiv:nnnn.nnnnn" is the right citation format? (To me it seems redundant to have the word "arXiv" twice.)

File: latex/acl_natbib.bst [hunk 1]
```diff
@@ -490,7 +492,7 @@ FUNCTION {emphasize}
     { "\emph{" swap$ * "}" * }
   if$
 }
-FUNCTION {tie.or.space.prefix}
```
Comment (davidweichiang) [inline latex/acl_natbib.bst:493]: preceeding -> preceding

Comment (LSinev) [reply latex/acl_natbib.bst:493]: fixed

**Comment by danielgildea** [comment]
1) I agree with David that it is redundant to have "arxiv" twice, and, poking around, I don't see other people doing this.

2) The .bst file has some comments saying that it is produced by a script.  We were maintaining the input to the script, but it seems to have disappeared from git.  Let's maintain this .bst file from now on, and remove the comment saying not to maintain it.

**Comment by LSinev** [comment]
1. I will remove unnecessary duplication
2. Not sure which comment need to be removed. I will try to reapply urlbst script if it will not make things worse

**Comment by LSinev** [comment]
One of commits creates some sort of "nourl" state.
I applied latest urlbst 0.9.1 after that.
Some lines have changes in trailing space removal (comparing with starting state of this PR).

Screenshot of result (zip archived files from PR opening message can still be used to check) 
image

**Comment by LSinev** [comment]
I think I can check if it will be better to recreate the whole file from the latest version of makebst and merlin.mbs (2011 instead of 2002 described in this file comments), selecting everything to comply with later changes described in comments inside.

**Comment by LSinev** [comment]
Rebuilt from scratch with `custom-bib` (options described in the commit message) and applied all the improvements that still make sense.

[acl_natbib_basic.zip](https://github.com/acl-org/acl-style-files/files/14546109/acl_natbib_basic.zip) — docstrip batch job (to be run with `latex acl_natbib_basic.dbj`) for archival purposes (probably in a repo with sources of style files).

Files to check and test applied changes.

[acl_natbib_new202403_test.zip](https://github.com/acl-org/acl-style-files/files/14546115/acl_natbib_new202403_test.zip)

[bib_experiment_after_recreation_with_improvement.pdf](https://github.com/acl-org/acl-style-files/files/14546116/bib_experiment_after_recreation_with_improvement.pdf)

[bib_experiment_before.pdf](https://github.com/acl-org/acl-style-files/files/14546117/bib_experiment_before.pdf)

**Comment by danielgildea** [comment]
Thanks for your work.  I've tested with some other papers, and this looks good to me.

**Pull request merged**